ladybug/legend.py

```diff
@@ -0,0 +1,877 @@
+        legend = Legend(data, LegendParameters(min=300, max=2000, number_of_segments=3,
+                                               ordinal_dictionary=ordinal_dict))
+        print(legend.segment_text)
+        print(legend.segment_mesh)
+        print(legend.segment_colors)
+        print(legend.value_colors)  # colors in between dict categories are interpolated
+        legend.legend_parameters.continuous_colors = False  # get data in only 3 colors
+        print(legend.value_colors)  # data colors align with number_of_segments
+
+        >> ['low', 'desired', 'too much']
+        >> Mesh3D (3 faces) (8 vertices)
+        >> ((R:75, G:107, B:169), (R:249, G:235, B:89), (R:234, G:38, B:0))
+        >> ((R:75, G:107, B:169), (R:75, G:107, B:169), (R:118, G:150, B:205),
+            (R:230, G:231, B:134), (R:234, G:38, B:0), (R:234, G:38, B:0))
+        >> ((R:75, G:107, B:169), (R:115, G:147, B:202), (R:115, G:147, B:202),
+            (R:115, G:147, B:202), (R:115, G:147, B:202), (R:234, G:38, B:0))
+
+        ##
+        data = [-0.5, 0, 0.5]
+        ordinal_dict = {-3: 'Cold', -2: 'Cool', -1: 'Slightly Cool', 0: 'Neutral',
+                        1: 'Slightly Warm', 2: 'Warm', 3: 'Hot'}
+        legend = Legend(data, LegendParameters(min=-1, max=1, number_of_segments=3,
+                                               ordinal_dictionary=ordinal_dict))
+        print(legend.segment_text)
+        print(legend.segment_mesh)
+        print(legend.segment_colors)
+        legend.legend_parameters.number_of_segments = 5
+        print(legend.segment_text)
+        legend.legend_parameters.min = -2
+        legend.legend_parameters.max = 2
+        print(legend.segment_text)
+        >> ['Slightly Cool', 'Neutral', 'Slightly Warm']
+        >> Mesh3D (3 faces) (8 vertices)
+        >> ((R:75, G:107, B:169), (R:249, G:235, B:89), (R:234, G:38, B:0))
+        >> ['Slightly Cool', '', 'Neutral', '', 'Slightly Warm']
+        >> ['Cool', 'Slightly Cool', 'Neutral', 'Slightly Warm', 'Warm']
+    """
+
+    def __init__(self, values, legend_parameters=None):
+        """Initalize Ladybug Legend.
+
+        Args:
+            values: A List or Tuple of numerical values that will be used to
+                generate the legend and colors.
+            legend_parameters: An Optional LegendParameter object to override
+                default parameters of the legend.
+            """
+        # check the inputs
+        assert isinstance(values, Iterable) \
+            and not isinstance(values, (str, dict, bytes, bytearray)), \
+            'values should be a list or tuple. Got {}'.format(type(values))
+        self._values = values
+        if legend_parameters is not None:
+            assert isinstance(legend_parameters, LegendParameters), \
+                'Expected LegendParameters. Got {}.'.format(type(legend_parameters))
+            self._legend_par = legend_parameters.duplicate()
+            self._is_legend_parameters_default = False
```

Comment:
This is helpful for the objects that possess the `Legend` as one of their properties (like the `ResultMesh` object).  Particularly when we call the `to_dict()` method, we know that, if the `LegendParameters` assigned are default, there's no need to write them into the dictionary (the default ones will just be re-assigned when the object is re-serialized).  Admittedly, this is not a critical feature and everything will still work if we always write explicit `LegendParameters` into the dictionary. However, if our goal with the dictionary serialization is to keep the dictionary description of the object as minimal as is necessary, this seems like the right way to do it.

Reply:
Generally speaking the logic of keeping it as small as possible makes sense. Since we won't send 1000s of LegendParameters I would say in this case being explicit is the better approach and the size is not the critical factor.

Reply:
Ok. I will take this out.